### DISCUSSION TITLE
Working Group: Code of Conduct

### DISCUSSION THREAD

**Discussion by ferdi05** [meilisearch/devrel#276] [General]
In September 2021, @bidoubiwa @ferdi05 and [Morgane Neff](https://www.linkedin.com/in/morgane-neff-8b1334102/) (HR manager at MeiliSearch) launched a Working Group. The objective of this group is to add a Code of Conduct where needed to MeiliSearch discussion spaces.

Here's the report from the first meeting. Any comments would be appreciated.

# 2021-09-21 - Kickoff

## Current situation at MeiliSearch

- There is no Code of Conduct (CoC) for our [Slack workspace](https://slack.meilisearch.com/), where we have many interactions with our community. We can't handle current Slack members and newcomers in the same way.
- There is a [CoC](https://github.com/meilisearch/MeiliSearch/blob/main/CODE_OF_CONDUCT.md) on the [MeiliSearch GitHub repo](https://github.com/meilisearch/MeiliSearch), but it's missing for many repos. And even if it's there, it's not easy to make sure that contributors are reading it before their first interactions on the repo (CoC applies to GitHub comments too).
- We wanted to review our CoC for a while now - see #186  and #187 

## Notes from the meeting

- CoC also applies to all MeiliSearch employees (Morgane Neff has to create an internal company policy).
- We need a specific contact person to make sure that the CoC is enforced. A general email like bonjour@meilisearch.com is not enough. And we could state that any MeiliSearch employee is allowed to enforce the CoC. It will be @tpayet , and we'll add his email address, Twitter and Slack handles.
- If the MeiliSearch team cannot unanimously come up with a decision about a behavior, @tpayet will make the call.
- We could create a document to tell the MeiliSearch employees and the community members what to do if they face unacceptable behavior.
- Wondering if there is an unacceptable behavior is already too much. We don't leave room for uncertainty.
- We need to look for a way to get our contributors to opt-in to the CoC when they are contributing, but this should be light enough not to prevent contributors from actually contributing. @bidoubiwa saw processes that are too complex with some other open-source projects. We may be able to customize the process by following.
- When registering as a new user to Slack, we could add an opt-in to the process.
- For existing Slack members, we could ask them to comply with the CoC before a specific date; otherwise, they won't be able to access our Slack workspace before they comply with our CoC.
- The current Code of Conduct may have been changed from the original one. And there is a [new version](https://www.contributor-covenant.org/version/2/1/code_of_conduct/). We should start from there.
- We could have different CoC versions customized to GitHub and Slack to adapt vocabulary. For instance, `project maintainers`, `commits`, `issues` do not apply to Slack.
- We could add a part at the CoC ending that tells what parts were modified compared to the original CoC template.

## Todo

- [ ]  Create a document to tell how to behave in case of unacceptable behavior
- [ ]  Put our CoC in every GitHub public repo (ask @curquiza if we can automate this)
- [ ]  Work on our version of the CoC
- [ ]  Publish this document on GitHub
- [ ]  Get some inspiration from the Open Source Guides [article on CoC](https://opensource.guide/code-of-conduct/), especially the Section 3 example

**Comment by ferdi05** [comment]
just saw the [CoC](https://github.com/devrelcollective/getting-started/blob/main/CodeOfConduct.md) for the DevRel Collective Slack community

**Comment by ferdi05** [comment]
I just read both the [Contributor Covenant CoC](https://www.contributor-covenant.org/version/2/1/code_of_conduct/) and the [DevRel Collective Slack community CoC](https://github.com/devrelcollective/getting-started/blob/main/CodeOfConduct.md).

They're both good.
The Contributor Covenant CoC is concise, and gives great details on Enforcement (see the `Enforcement Guidelines` section). 
The DevRel Collective CoC gives great details on what to say/share and what to avoid talking about.

So I don't know what to think, if we should choose only one of them (with the risk of lacking details), adapt one of them with what's good coming from the other CoC (it can turn awful), or have a different CoC for GitHub and Slack (brings confusion, could be over-thinking the problem).

I would be happy to know what are your thoughts.

**Reply by laurajwilkinson** [reply]
**tl;dr**: I think that in order to be effective, we need to focus on challenging lower-level inappropriate behaviour to show that we are serious about maintaining professional boundaries. If we agree on this, we could take inspiration from advice for children on handling bullies - the core of which is a simple, assertive challenge such as "that wasn’t funny", "that’s not cool", "stop it", "that's mean". Such responses benefit from role-play and rehearsal in order to be able to apply them quickly when needed. Someone whose inappropriate behaviour persists despite this is committing a larger offence and as such needs to be dealt with by an authority greater than Meili.

**Longer response**: My viewpoint is influenced by the case of Jon Tennant, who was a high-profile member of the open science community. He was banned from the event OpenCon but was not told why, and there was no process or way to appeal (more info at [https://www.change.org/p/open-con-oppose-a-lack-of-accountability-a-call-for-answers](https://www.change.org/p/open-con-oppose-a-lack-of-accountability-a-call-for-answers).) Several months later, he died in an unrelated accident, his reputation in ruins. His sister is campaigning for accountability ([https://rebeccatennant10.wordpress.com/2020/05/01/what-really-happened-jon-tennant/](https://rebeccatennant10.wordpress.com/2020/05/01/what-really-happened-jon-tennant/)) but since he is dead, no-one really wants to listen.

The way Jon was treated made me realise how although codes of conduct have a noble aim, their enforcement must be accountable, based on evidence, and limited to brief sanctions - anything more serious needs to be handled by proper legal mechanisms.

1. Consider renaming this policy (example: professional conduct/behaviour policy), to encourage people to read it (there is such a proliferation of Codes of Conduct that people assume they’ve read the same thing elsewhere), and to avoid the unfortunate acronym "CoC".

2. “CoC also applies to all MeiliSearch employees (Morgane Neff has to create an internal company policy).” - Could we aim for one policy for all people and channels, all Meili interactions?

3. “We need a specific contact person to make sure that the CoC is enforced.” Suggest that the Code is a model of positive behaviour but that warnings and sanctions may follow. Serious breaches will be referred to legal authorities (French?) 

4. “And we could state that any MeiliSearch employee is allowed to enforce the CoC.” Perhaps all employees could be a point of contact, but perhaps enforcement needs to be done by those with suitable training? (unless we plan to include all staff in this)

5. “regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.” Many of these factors come from country-specific legislation, and any breaches of such law should be referred to the relevant authorities, rather than us taking these matters into our own hands. I think we could end the sentence at “a harassment-free experience for everyone” without loss of meaning or positive intention.

6. “Using welcoming and inclusive language” is quite vague and subjective - suggest “Being welcoming and listening to each other” (also puts the emphasis on behaviour, not just words)

7. “We could create a document to tell the MeiliSearch employees and the community members what to do if they face unacceptable behavior.” This is a great idea - give practical steps and perhaps include wording or phrases people can use, as without practice, it can be hard to know what to say when an incident happens. It could be helpful to rehearse intervening and applying the Code

8. “unacceptable behavior” - state clearly what this means (as briefly as possible). Could we start from a list of desired behaviours, and then it follows that behaviour is “unacceptable” when it does not conform? Also need to state that a serious breach will be referred to the official legal mechanism (as we can’t play judge and jury for this) and state legal jurisdiction (France?)

9. From [https://www.contributor-covenant.org/version/2/1/code_of_conduct/ ](https://www.contributor-covenant.org/version/2/1/code_of_conduct/) Standards and Enforcement sections are good models, but suggest enforcement limited to correction, warning, fixed-term ban. A permanent ban leaves no room for rehabilitation, and it is all too easy to come back in under a different username, so I suggest we drop that sanction.

10. Maybe only take action based on tangible evidence such as Github traces or screenshots, to avoid unfounded accusations? What do we do in cases where it’s one person’s word against another?

**Reply by ferdi05** [reply]
Thanks so much @laurajwilkinson for your feedback. I'm sorry for not replying earlier, I guess that I got scared of those 10 points 😅 as I already had a lot on my plate. This is still true, but this topic needs to progress.

I realize that the [Contributor Covenant Code of Conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct/) was not the right one. This is now fixed.

Thanks for letting me know about the Jon Tennant case. I was not aware of this, and this is indeed quite shocking.
I agree that we need tangible evidence as suggested in item 10:

> 10. Maybe only take action based on tangible evidence such as Github traces or screenshots, to avoid unfounded accusations? What do we do in cases where it’s one person’s word against another?

We can't let ourselves be opaque. Our decisions should be evidence-based and we will be accountable for them.

I agree with item 3, for anything but brief and limited sanctions, we should leave it to an authority greater than Meili.

> 3. “We need a specific contact person to make sure that the CoC is enforced.” Suggest that the Code is a model of positive behaviour but that warnings and sanctions may follow. Serious breaches will be referred to legal authorities (French?)


Now let's get to the item list:

> 1. Consider renaming this policy (example: professional conduct/behaviour policy), to encourage people to read it (there is such a proliferation of Codes of Conduct that people assume they’ve read the same thing elsewhere), and to avoid the unfortunate acronym "CoC".

Even if I agree, there is an issue with GitHub: it won't consider it as a Community health file if we give the file a title different from `CODE_OF_CONDUCT.md` (see [here](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file)).
To preserve consistency, we should call this Code of Conduct everywhere. Also, I'd like to know what is wrong with using `CoC`(I'm not sure of the capitalization).

> 2. “CoC also applies to all MeiliSearch employees (Morgane Neff has to create an internal company policy).” - Could we aim for one policy for all people and channels, all Meili interactions?

I wish this was possible. The main issue I see is that it seems too hard to find a wording that suits in-person events, digital events, asynchronous discussion platforms, and the work environment.

> 4. “And we could state that any MeiliSearch employee is allowed to enforce the CoC.” Perhaps all employees could be a point of contact, but perhaps enforcement needs to be done by those with suitable training? (unless we plan to include all staff in this)

Agreed, this is too vague.

> 5. “regardless of age, body size, disability, ethnicity, sex characteristics, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.” Many of these factors come from country-specific legislation, and any breaches of such law should be referred to the relevant authorities, rather than us taking these matters into our own hands. I think we could end the sentence at “a harassment-free experience for everyone” without loss of meaning or positive intention.

I'm not a law specialist, so I wonder how sure you are about this being a breach of law everywhere (or France only). I like the idea of having a list as no one can claim that the issue was not listed and that the Code of Conduct was too vague.

> 6. “Using welcoming and inclusive language” is quite vague and subjective - suggest “Being welcoming and listening to each other” (also puts the emphasis on behaviour, not just words)

This does not exist anymore in v2.1 of the  [Contributor Covenant Code of CONDUCT](https://www.contributor-covenant.org/version/2/1/code_of_conduct/). But indeed this is in our current [Code of Conduct](https://github.com/meilisearch/MeiliSearch/blob/main/CODE_OF_CONDUCT.md).

> 7. “We could create a document to tell the MeiliSearch employees and the community members what to do if they face unacceptable behavior.” This is a great idea - give practical steps and perhaps include wording or phrases people can use, as without practice, it can be hard to know what to say when an incident happens. It could be helpful to rehearse intervening and applying the Code

👌

> 8. “unacceptable behavior” - state clearly what this means (as briefly as possible). Could we start from a list of desired behaviours, and then it follows that behaviour is “unacceptable” when it does not conform? Also need to state that a serious breach will be referred to the official legal mechanism (as we can’t play judge and jury for this) and state legal jurisdiction (France?)

I find it hard to create a list, this would be so long and confused, and we would miss many situations.

> 9. From https://www.contributor-covenant.org/version/2/1/code_of_conduct/ Standards and Enforcement sections are good models, but suggest enforcement limited to correction, warning, fixed-term ban. A permanent ban leaves no room for rehabilitation, and it is all too easy to come back in under a different username, so I suggest we drop that sanction.

Agreed.

So this is the end of this very long message. If you feel like it needs a lengthy response, I would suggest to create a small recap and to discuss this with other Working Group members.

**Reply by laurajwilkinson** [reply]
Some useful ideas for drafting policy/governance: [A Practical Framework for Applying Ostrom’s Principles to Data Commons Governance](https://foundation.mozilla.org/en/blog/a-practical-framework-for-applying-ostroms-principles-to-data-commons-governance/) see "Table 2: Design questions for data commons" for some very good questions to ask

**Comment by ghost** [comment]
Hello all,

Thanks for all the input and feedback.

I tried my hand at writing the first draft of a revamped Code of Conduct. I did my best to utilise the previous code as well as the feedback shared so far. I don't think I managed to include everything, but I hope I coved enough bases while still maintaining a certain level of flexibility.

Please feel free to have a read-through and share your thoughts, if there's anything else you think is essential for me to add, don't hesitate to flag it:
(tagging the people from the original working group: @ferdi05 @bidoubiwa @morganeff)
# First draft CoC

Meilisearch, and all its employees, is an entity fully committed to building an inclusive, encouraging, inspiring and diverse space for everyone that may want to join our community or our project. Our number one priority is ensuring you have fun and contribute in a way that makes sense to you while navigating in a safe and respectful environment.

The community and all contributors are at the core of what we do. Thus it is paramount for Meilisearch that the members feel supported. Our Code of Conduct is an extension of our company vision and values.

In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to make participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity, and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.

Any community member or contributor evolving in one of Meilisearch spaces must be aware of the Code of Conduct and uphold its values.

### Who will receive any reports of the breach of the Code of Conduct?

If you are a victim or a witnesses of a breach of the Code of Conduct, we kindly request you reach out to bonjour@meilisearch.com so that we can start an official investigation based on your feedback. We know the situations that may have occurred can be painful to retell. Still, we would be very grateful if you could share as many details or any physical evidence, which would greatly facilitate the investigation. In order to protect the people involved in the investigation, we must maintain anonymity throughout the process. Details of said investigation won't be shared, however, we may communicate separately concerning next steps or sanctions if necessary without sharing any personal or sensitive information.

If your complaint is addressed toward a Meilisearch employee that you fear may be a recipient of bonjour@meilisearch.com, please feel free to use (ALTERNATIVE POINT OF CONTACT) instead.

We take your words very seriously. Any abuse of this trust made in a malicious spirit or bad faith could be sanctioned.

### Scope

This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community. Examples of representing a project or community include using an official project email address, posting via an official social media account, or acting as an appointed representative at an online or offline event. Representation of a project may be further defined and clarified by project maintainers.

### Code of Conduct

➡️ Kindness: We all learn from each other and often communicate in a language that isn’t our mother tongue, with no visual or vocal cues to support our messages. Let’s look at each other with kindness when communicating together. We wish to foster a welcoming space where newcomers have a chance to find their own spot and where developers of all backgrounds and levels can support and elevate each other.

➡️ 0 tolerance policy: any form of physical, mental, sexual abuse, harassment, or toxicity occurring on one of Meilisearch safe spaces (our various social media, Github, Slack, conventions, job interviews, etc.) will be met with immediate removal from said spaces and a permanent ban. It is perfectly acceptable to disagree with each other, it’s indeed one of the best ways to improve ourselves and make our experiences grow. However, disagreement must be handled in a mature manner that allows all participants to maintain their safety and dignity.

- Do you feel you’re losing control of a situation or an interaction? Take a deep breath, take a step back, go for a walk, ask for a mediator (anyone from the DevRel team: @ferdi05, @CaroFG, @oluademola, @Luna-meili or our HR team: @morganeff) to help smooth the conversation over. It’s natural that emotions run high at times, but don’t let them take over you.

➡️ Acceptance of each other: The Internet and online communities allow us to connect with people from all over the world and meet people we would never have had the chance to meet otherwise. This also means we will be confronted with world views very different from our own and what is familiar to us. It is essential for Meilisearch that we’re all respectful of each others’ cultures and identities. Discrimination based on nationality, ethnicity, sexual orientation, etc. will be punished accordingly to the severity of the action, to be determined through our investigation. This principle must coexist with the one around preserving each other safety, so topics such as transphobia, homophobia, ableism, ageism, colourism, etc. do not fall within the remit of this rule.

➡️ Lawful: The safety of our community and our employees is one of Meilisearch highest priorities. As such, we will not tolerate any form of unlawful behaviour on our channels. We will report any illegal behaviour to the appropriate authority of the platform on which the behaviour was perpetuated. Educate yourself on your local laws to respect them and report any illegal or suspicious activities, especially if they can put someone in danger.

➡️ Security: We strongly advise our community members and contributors against sharing personal or account information with anyone else, whether the people are present online or in real life. This includes information that could be used to identify you, like your telephone number, home address, etc., as well as information that could be used to compromise your account, like your email address or answers to your security questions, etc. Meilisearch employees will never ask you for your password. Please help us help you, it is also your responsibility to keep you and your accounts safe.

➡️ We’re in your corner: Don’t feel like you’re alone, Meilisearch employees are also there to roam through our various platforms and ensure the Code of Conduct is enforced. To protect our communities, representatives, and partners, we may remove any and all content published by you that breaches this Code of Conduct or our Terms of Use. The online world is a big place, though, so if you feel like we missed something important, do not hesitate to bring it to our attention

➡️ Was it a breach of the Code of Conduct? Situations aren’t always clear cut, and we can sometimes wonder whether or not the Code of Conduct was breached. Was it an aggressive argument or simply a passionate debate? Was someone being harassed, or is it an inside joke? It is always better to report any ambiguous situation rather than leave someone in a potentially vulnerable position. We will be grateful for your report and will investigate ourselves.

➡️ Meilisearch’s responsibility: Meilisearch cannot be held responsible for the actions or comments made by contributors or community members in the various online and social media outlets. Meilisearch reserves the right to remove any and all content published by a contributor without prior warning or notification. However, Meilisearch and its employees vow to do do their utmost best to foster a positive environment that complies with this Code of Conduct.

➡️ Sanctions: following an investigation that will be led internally, the person(s) in charge of the investigation will decide on a sanction based on the gravity of the offense. All individuals involved in a report will be contacted as to allow everyone to offer their side of the conflict and to defend themselves. The final decision won’t be taken unilaterally by a single person. If a consensus cannot be reached, the final call will be made by the acting CEO of Meilisearch, currently @qdequele. Here are a few example of sanctions that can be taken:

- it is possible no action will be taken for lack of proof or facts
- written warning
- temporary ban from our online spaces
- a permanent ban from our spaces
- legal action is the offense is of illegal nature

**Reply by ferdi05** [reply]
Congrats for this amazing work Luna, that's really concise and precise, and I can't think of anything that I feel like is missing.

I'm wondering: have you used any inspiration? Possibly one or more existing CoC templates?

> Any community member or contributor evolving in one of Meilisearch spaces must be aware of the Code of Conduct and uphold its values.

We need to ensure this somehow. For instance, I'd like new members of the community to `agree` with the CoC before they can join any channel. And we should think of something for existing members.   Also, here you’re talking about the Meilisearch spaces… which is vague. We should maybe be more explicit about it. Later you're using `one of Meilisearch safe spaces (our various social media, Github, Slack, conventions, job interviews, etc.)`
And once it’s done, how do we apply the CoC to GitHub, Twitter, … ?

bonjour@meilisearch.com => that’s a generic address that many people receive. First of all, I’d like us to create a new alias that looks dedicated to this (something like cocbreach@meilisearch) and be really specific about who receives this, and who is responsible for taking decisions here (can be @tpayet or @qdequele ultimately). Later you're saying `by one of the founder of Meilisearch`I would go further and name them explicitly, and possibly name one among them to make the call.

> Further details of specific enforcement policies may be posted separately.

I don't get this, can you explain?

> ask for a mediator to help smooth the conversation over

Who are mediators?

> Discrimination based on nationality, ethnicity, sexual orientation, etc. will be punished with a removal from Meilisearch spaces

Why not talking about sanctions here? We don't really know what the extent of the sanction for this. Also you're somehow setting the limit of the sanction, not allowing it to be larger.

> report any illegal or suspicious activities

Who to report to?

> against sharing personal or account information with anyone else

Who is anyone else?

> legal action iF the offense is of illegal nature

**Reply by ghost** [reply]
Thanks very much for your feedback!

> I'm wondering: have you used any inspiration? Possibly one or more existing CoC templates?

No specific basic template, I used a mix of old CoC I liked and included points that were relevant for Meilisearch.

> We need to ensure this somehow. 

At this point in time, there's no way to automate such a process on Slack for instance, I investigated this with the Slack team. In terms of ensuring people agree with the CoC, we can write on our social media (eg: Twitter, LinkedIn, Reddit) something like "by interacting with our posts and our account, you agree to comply with our CoC" and link it for people to be able to read. But otherwise there's no proper way to block people from channels. I believe it's a possibility on Discord.


> I’d like us to create a new alias that looks dedicated to this (something like cocbreach@meilisearch)

Agreed

> Later you're saying by one of the founder of MeilisearchI would go further and name them explicitly, and possibly name one among them to make the call.

Agreed, needs to be defined with Morgane and them. In the old CoC it was @tpayet, I spoke with him and he said it might be more interesting for the final decision to be taken by the acting CEO of Meilisearch, so in this case, it would be @qdequele 

> I don't get this, can you explain?

This is one of the parts from the old CoC, I believe that this refers to the fact we need to maintain confidentiality about the details of a case, but we might communicate on the sanction chosen without sharing details about the core issue.

> Who are mediators?

Personally, I was thinking anyone from the DevRel or HR team to help navigate the conversation

>Why not talking about sanctions here? 

The sanction would be the removal from our spaces (eg: removal from the Slack or being blocked), basically the sanction is equivalent to a ban.

>Who to report to?

Depends where the illigal activity is happening, if it's Twitter we must report them to Twitter, if it's Slack we need to contact the Slack team etc

> Who is anyone else?

Any other user on the internet or in real life

Hope my answer made sense, if not please don't hesitate to let me know!

**Reply by ferdi05** [reply]
Thanks for your quick reply @Luna-meili 
I don't have much to say, this looks pretty good to me. When I asked for more details, I was meaning that this should be added to CoC to disambiguate any situations.

Now should we collect feedback from @morganeff and @bidoubiwa ?

Some remaining comments:

>>  I don't get this, can you explain?
> 
> This is one of the parts from the old CoC, I believe that this refers to the fact we need to maintain confidentiality about the details of a case, but we might communicate on the sanction chosen without sharing details about the core issue.

Then I think that we should rewrite it to make it clearer.

>> Who are mediators?
> 
> Personally, I was thinking anyone from the DevRel or HR team to help navigate the conversation

Ok with this, it would be great to settle who this persons are


>> Why not talking about sanctions here?
> 
> The sanction would be the removal from our spaces (eg: removal from the Slack or being blocked), basically the sanction is equivalent to a ban.

I still don't get why we should already say what sanction it would be. It should be left in doubt, because I guess in the worst case we may want to increase the sanction.


>> Who to report to?
> 
> Depends where the illigal activity is happening, if it's Twitter we must report them to Twitter, if it's Slack we need to contact the Slack team etc

Then this should be said somehow.

**Reply by ghost** [reply]
Updated all the points you mentioned, I sent the CoC to @morganeff and @bidoubiwa for their consideration if/when they have the time.

The main outstanding point is about bonjour@meilisearch.com, we might need HR point of view for this one. : )

**Reply by bidoubiwa** [reply]
Hey! Incredible work <3 I think everything is pretty clear and exhaustif. 
I would like to add one point, but it may not be useful. 

> Was it an aggressive argument or simply a passionate debate? Was someone being harassed, or is it an inside joke?

Often, no harm is meant by someone making a discriminating joke or comment. It may be something accepted in their environment, and thus they might not understand why we get back to them about their behavior. Should we mention that, even when the intentions are innocent, it does not excuse discrimination? 

If it was already said before, my bad!

**Comment by morganeff** [comment]
Hey! thanks @Luna-meili for all this work. It's really complete. 

However I would make this more synthetic and straight-to-the-point. Since a code of conduct isn't "fun" to read, but is necessary, we should try making this shorter. My suggestions:
- I would take out sentences that aren't necessary, or try to make them shorter
- Maybe change the structure to offer clarity. Maybe something like: Why we need this code / 0 tolerance policy and sanctions / Main principles (don't share personal information, treat each other with kindness, etc.) / What to do if you witness a breach of conduct / What happens on Meili's side (who is involved, how we lead an investigation)
- Make it more straightforward. This is a personal opinion, but it's ok to me if this doesn't sound "friendly". Its purpose is to let people know we will be firm in our assessment and sanctions, so to me it should be written this way. 

Here are my other comments for the current version:

### First draft of CoC

> Any community member or contributor evolving in one of Meilisearch spaces must be aware of the Code of Conduct and uphold its values.

I agree this should be something that is ensured in some way. Also, in this phrase, maybe we could make it more straightforward that if someone is evolving in one of the Meilisearch spaces, we will assume they have taken knowledge of the CoC.

### Who will receive any reports of the breach of the Code of Conduct?

I would create two sections: 

- one about who at Meilisearch is in charge of addressing emails and allegations about breaches of conduct, who launches the investigation, and who takes the decision about sanctions
- another section about what to do if you're a victim or a witness, so people easily know straight away what they have to do.

Maybe I would stress the fact that if there's a doubt, there's no doubt? for people who might be disposed to convince themselves that they're exaggerating and they're imagining things. My point is to be reassuring about the fact that this is important for us, and they shouldn't feel like they're bothering us or anything.

>If you are a victim or a witnesses of a breach

If you are a victim of a breach `or witness a breach` in the CoC...

>we kindly request you reach out to [bonjour@meilisearch.com](mailto:bonjour@meilisearch.com)

This address makes me feel like I'd be writing an email to robots. Can we create a specific address? 

>Any abuse of this trust made in a malicious spirit or bad faith could be sanctioned.

Yes, very good point. I would say `will` instead of `could`, though. 

### Scope

>This Code of Conduct applies both within project spaces and in public spaces when an individual is representing the project or its community

For me (but maybe it's because I'm outside of the open-source space, so I'm missing some notions), this definition needs clarification.
`project spaces and in public spaces` -> this doesn't have any meaning to me, maybe we should make it more specific? Also, I don't know if this is implied, but we should say something about the fact that this CoC applies to virtual and physical spaces 
`an individual` -> who? someone from Meili? 
`the project or its community` is quite vague, it needs clarification.

### Code of Conduct

**Kindness**

>Let’s look at each other with kindness when communicating together

Would change to `let's consider` each other with kindness

**0 tolerance policy**

>...or toxicity occurring on one of Meilisearch safe

`On` -> `in`

100% agree with the 0 tolerance policy, but I think we should make this a separate section so it stands out. It's probably the most important thing to know for people reading this CoC.

>Meilisearch safe spaces

Why not just `Meilisearch spaces`?

>disagreement must be handled in a mature manner

mature manner -> respectful manner?

>It’s natural that emotions run high at times, but don’t let them take over you.

I would erase this sentence, or change the "don't let them take over you" into something less directive.

**Acceptance of each other**

Change the title to `Tolerance`?

>will be punished accordingly to the severity of the action

I find this a bit vague. Also we speak of a 0 tolerance policy: shouldn't this be punished by immediate ban? 

**Lawful**

Lawfulness

**Security**

I would change this to `Personal information`

**We're in your corner**

>Don’t feel like you’re alone

I would take out this part of the sentence. 

I would change the paragraph actually:

Meilisearch employees are here to navigate and examinate our various platforms to make sure the Code of Conduct is enforced. To protect our communities, representatives, and partners, we will remove any content that represents a breach in this Code or our Terms of Use. However, if you feel like we missed something important, do not hesitate to bring it to our attention. 

**Meilisearch’s responsibility**

>However, Meilisearch and its employees vow to do do their utmost best to foster a positive environment that complies with this Code of Conduct.

Would take this out.

**Sanctions**

I would maybe make this a separate section. 
I would also reorganize the paragraph so it's clearer:

In the case of an investigation, all parties involved will be given the opportunity to express themselves and give their version of the events. The Meilisearch employees in charge of the investigation will collectively decide on the sanction. If a consensus cannot be reached, the final call will be made by the acting CEO of Meilisearch, currently @qdequele. In the case of a lack of proof or facts, it may not be possible to take action. 

Sanctions may include but will not be limited to:

- a written warning
- a temporary ban from our online spaces
- a permanent ban from our spaces
- legal action is the offense is of illegal nature

**Reply by ghost** [reply]
Thank you very much for the comprehensive feedback! I restructured the CoC following your suggestions and did my best to make it more concise. I could be shorter (especially in the first half) but I started thinking maybe I was cutting too much out. Happy to hear your opinion on that. I know commenting on GitHub on long texts can be difficult, so I also created the document on [Notion](https://www.notion.so/meilisearch/Code-of-conduct-of-slack-3d7edc11f7f9462582925112f7f276cf) _(Private link)_ if it makes the feedback process easier.

Here's the result following the implementation of your feedback:

# Draft CoC

Meilisearch is an entity fully committed to building an inclusive, encouraging, inspiring and diverse space for everyone that may want to join our community or our project. The community and all contributors are at the core of what we do. Thus it is paramount for Meilisearch that the members feel supported. Our Code of Conduct is an extension of our company vision and values.

In the interest of fostering an open and welcoming environment, we as contributors and maintainers pledge to make participation in our project and our community a harassment-free experience for everyone, regardless of age, body size, disability, ethnicity, sex characteristics, gender identity, and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation.

It will be assumed that any community member or contributor evolving in one of Meilisearch spaces is aware of the Code of Conduct and agreed to uphold its values. 

### Scope

This Code of Conduct applies both to virtual (GitHub, Twitter, Slack, LinkedIn, and any other social media) and physical spaces (interviews, conventions, events, workshops, etc) when an employee or a contributor is representing the company, one of its project, or the community.  Representation of a project may be further defined and clarified by project maintainers.

### 0 tolerance policy

Any form of physical, mental, sexual abuse, harassment, or toxicity occurring in one of Meilisearch spaces (our various social media, Github, Slack, conventions, job interviews, etc. as mentioned above) will be met with immediate removal from said spaces and a permanent ban. Disagreements must be handled in a respectful manner that allows all participants to maintain their safety and dignity.
It is essential for Meilisearch that we’re all respectful of each others’ cultures and identities. Discrimination based on nationality, ethnicity, sexual orientation, etc. will not be tolerated. Respect of each other cultures must coexist with the principle of preserving each other safety, so topics such as transphobia, homophobia, ableism, ageism, colourism, etc. do not fall within the remit of this rule.

### Sanctions

In the case of an investigation, all parties involved will be given the opportunity to express themselves and give their version of the events. In the case of a lack of proof or facts, it may not be possible to take action.

Sanctions will be enforced immediately and may include but will not be limited to:

- a written warning
- a temporary ban from our online spaces
- a permanent ban from our spaces
- legal action is the offense is of illegal nature

We take your words very seriously. Any abuse of this trust made in a malicious spirit or bad faith could be sanctioned.

### Code of Conduct

➡️ Kindness: We often communicate in a language that isn’t our mother tongue, with no visual or vocal cues to support our messages. Let’s consider at each other with kindness when communicating together to foster a welcoming space for newcomers and veterans alike. 

➡️ Lawfulness: Will not tolerate any form of unlawful behaviour on our channels. We will report any illegal behaviour to the appropriate authority of the platform on which the behaviour was perpetuated. Educate yourself on your local laws to respect them and report any illegal or suspicious activities.

➡️ Personal information: We strongly advise our community members and contributors against sharing personal or account information with anyone else, whether the people are present online or in real life. This includes information that could be used to identify you, like your telephone number, home address, etc., as well as information that could be used to compromise your account, like your email address or answers to your security questions, etc. 

➡️ We’re in your corner: Meilisearch employees are here to navigate and examine our various platforms to make sure the Code of Conduct is enforced. To protect our communities, representatives, and partners, we will remove any content that represents a breach in this Code or our Terms of Use. However, if you feel like we missed something important, do not hesitate to bring it to our attention.

➡️ If there’s a doubt, there’s no doubt: Situations aren’t always clear cut, and we can sometimes wonder whether or not the Code of Conduct was breached. It is always better to report any ambiguous situation rather than leave someone in a potentially vulnerable position. We will be grateful for your report and will investigate ourselves.

➡️ Meilisearch’s responsibility: Meilisearch cannot be held responsible for the actions or comments made by contributors or community members in the various online and social media outlets. Meilisearch reserves the right to remove any and all content published by a contributor without prior warning or notification. 

### Process of reporting a breach of the Code of Conduct?

- If you are a victim or a witness a breach of the Code of Conduct:

Reach out as soon as possible to [bonjour@meilisearch.com](mailto:bonjour@meilisearch.com) so that we can start an investigation based on your feedback. We’d be very grateful if you could share as many details or any physical evidence, which would greatly facilitate the investigation. 

If your complaint is addressed toward a Meilisearch employee that you fear may be a recipient of [bonjour@meilisearch.com](mailto:bonjour@meilisearch.com), please feel free to use (ALTERNATIVE POINT OF CONTACT) instead.

- Who will lead the investigation:

The HR (@of team members) and DevRel team (@of team members) will lead the investigation. A decision on the sanction will be taken based on the gravity of the offense. If a consensus cannot be reached by the investigation team, the final call will be made by the acting CEO of Meilisearch, currently @qdequele.

NB: To protect the people involved in the investigation, we must maintain anonymity throughout the process. Details of said investigation won't be shared, however, we may communicate separately concerning next steps or sanctions if necessary without sharing any personal or sensitive information.
